src/main/java/org/sonarlint/intellij/config/project/AddEditExclusionDialog.java

```diff
@@ -0,0 +1,232 @@
+        updateOk();
+      }
+    };
+
+    fileTextField.getTextField().getDocument().addDocumentListener(docListener);
+    directoryTextField.getTextField().getDocument().addDocumentListener(docListener);
+    globTextField.getDocument().addDocumentListener(docListener);
+
+    ActionListener listener = e -> updateControls();
+    directoryRadioButton.addActionListener(listener);
+    globRadioButton.addActionListener(listener);
+    fileRadioButton.addActionListener(listener);
+    updateControls();
+  }
+
+  @NotNull
+  @Override
+  protected Action[] createActions() {
+    return new Action[] {getOKAction(), getCancelAction(), getHelpAction()};
+  }
+
+  @Override
+  protected void doHelpAction() {
+    HelpManager.getInstance().invokeHelp("ignoreUnversionedFilesDialog");
+  }
+
+  private void updateControls() {
+    directoryTextField.setEnabled(directoryRadioButton.isSelected());
+    globTextField.setEnabled(globRadioButton.isSelected());
+    fileTextField.setEnabled(fileRadioButton.isSelected());
+    helpLabel.setEnabled(globRadioButton.isSelected());
+
+    updateOk();
+  }
+
+  private void updateOk() {
+    boolean valid = StringUtils.trimToNull(getSelectedText().getText()) != null;
+    myOKAction.setEnabled(valid);
+  }
+
+  private JTextField getSelectedText() {
+    if (directoryRadioButton.isSelected()) {
+      return directoryTextField.getTextField();
+    } else if (fileRadioButton.isSelected()) {
+      return fileTextField.getTextField();
+    } else {
+      return globTextField;
+    }
+  }
+
+  @Nullable
+  protected JComponent createCenterPanel() {
+    return panel;
+  }
+
+  @CheckForNull
+  public ExclusionItem getExclusion() {
+    if (directoryRadioButton.isSelected()) {
+      return new ExclusionItem(DIRECTORY, relative(directoryTextField.getText()));
+    } else if (fileRadioButton.isSelected()) {
+      return new ExclusionItem(FILE, relative(fileTextField.getText()));
+    } else {
+      return new ExclusionItem(GLOB, relative(globTextField.getText()));
+    }
+  }
+
+  public void setExclusion(@Nullable ExclusionItem item) {
+    setTitle("Edit SonarLint File Exclusion");
+
+    if (item != null) {
+      switch (item.type()) {
+        case DIRECTORY:
+          directoryTextField.setText(item.item());
+          directoryRadioButton.setSelected(true);
+          break;
+        case FILE:
+          fileTextField.setText(item.item());
+          fileRadioButton.setSelected(true);
+          break;
+        case GLOB:
+          globTextField.setText(item.item());
+          globRadioButton.setSelected(true);
+          break;
+      }
+      updateControls();
+    }
+  }
+
+  private void createUIComponents() {
+    patternHelp = new HyperlinkLabel("");
+    Icon infoIcon = SonarLintIcons.INFO;
+    patternHelp.setIcon(infoIcon);
+    //hoverHyperlinkLabel1.setUseIconAsLink(true);
+    patternHelp.addMouseListener(new MouseAdapter() {
+      public void mouseEntered(MouseEvent e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)